File: src/com/codingbaby/ohmyidea/action/MotionAndInsertAction.java
package com.codingbaby.ohmyidea.action;

import com.codingbaby.ohmyidea.KeyHandler;
import com.codingbaby.ohmyidea.helper.EditorHelper;
import com.intellij.openapi.actionSystem.DataContext;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.editor.actionSystem.EditorAction;
import com.intellij.openapi.editor.actionSystem.EditorActionHandler;

/**
 *
 *
 */
public class MotionAndInsertAction extends EditorAction {

    public MotionAndInsertAction(){
        super(new EditorActionHandler() {
            @Override
            public void execute(Editor editor, DataContext dataContext) {
                int oldOffset = editor.getCaretModel().getOffset();
                editor.getCaretModel().moveToOffset(oldOffset + 1);
                EditorHelper.scrollCaretIntoView(editor);
                KeyHandler.toInsertMod();
            }
        });
    }
}

File: src/main/java/com/codingbaby/ohmyidea/action/MotionAndInsertAction.kt
package com.codingbaby.ohmyidea.action

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.Caret
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.editor.actionSystem.EditorAction
import com.intellij.openapi.editor.actionSystem.EditorActionHandler
import com.codingbaby.ohmyidea.KeyHandler
import com.codingbaby.ohmyidea.helper.EditorHelper


class MotionAndInsertAction : EditorAction(object : EditorActionHandler() {
    override fun doExecute(editor: Editor, caret: Caret?, dataContext: DataContext) {
        val oldOffset = editor.caretModel.offset
        editor.caretModel.moveToOffset(oldOffset + 1)
        EditorHelper.scrollCaretIntoView(editor)
        KeyHandler.toInsertMod()
    }
})
